apps/web/src/app/[locale]/challenge/[slug]/getChallengeRouteData.ts

```diff
@@ -2,12 +2,23 @@ import { type Session } from '@repo/auth/server';
 import { prisma } from '@repo/db';
 import { cache } from 'react';
 import { getAllFlags } from '~/utils/feature-flags';
+import { AOT_CHALLENGES } from './aot-slugs';
+import { daysAfterDecemberFirst } from '~/utils/aot';
 
 export type ChallengeRouteData = NonNullable<Awaited<ReturnType<typeof getChallengeRouteData>>>;
 
 // this is to data to populate the description tab (default tab on challenge page)
 export const getChallengeRouteData = cache(async (slug: string, session: Session | null) => {
   const featureFlags = await getAllFlags();
+  const isAotChallenge = AOT_CHALLENGES.includes(slug);
+
+  if (featureFlags.enableHolidayEvent && isAotChallenge) {
+    const [, day = '1'] = slug.split('-');
+    const daysPassed = daysAfterDecemberFirst();
+    if (parseInt(day) > daysPassed + 1) {
+      throw new Error('Not Available');
```

Comment:
lgtm, alternatively could return `null` and then `notFound()` in the route

Reply:
for sure, went with throw just because i was using `findFirstOrThrow` below. so just implicitly trusted we do something good haha